compiler/rustc_parse/src/validate_attr.rs

```diff
@@ -198,12 +157,85 @@ fn is_attr_template_compatible(template: &AttributeTemplate, meta: &ast::MetaIte
     }
 }
 
+pub fn check_attribute_safety(
+    features: &Features,
+    psess: &ParseSess,
+    safety: AttributeSafety,
+    attr: &Attribute,
+) {
+    if features.unsafe_attributes {
```

Comment:
Early returns help keep the indentation levels under control:
```suggestion
    if !features.unsafe_attributes {
        return;
    }
```

Ignore this comment unless I find anything useful to change.

Reply:
And it was already present in the existing code...